src/imputation/tmi_imputation.py

```diff
@@ -464,16 +472,16 @@ def run_tmi(
     TMILogger.info("Imputing for R&D type (civil or defence).")
     df = impute_civil_defence(df)
 
-    TMILogger.info("Calculating the trimmed mean for target variables")
-    df = tmi_pre_processing(df, target_variables)
+    lf_target_variables = config["imputation"]["lf_target_vars"]
+    df = tmi_pre_processing(df, lf_target_variables)
 
-    mean_dict, qa_df = create_mean_dict(df, target_variables)
+    mean_dict, qa_df = create_mean_dict(df, lf_target_variables, config)
 
     qa_df.set_index("qa_index", drop=True, inplace=True)
 
     qa_df = qa_df.drop("trim_check", axis=1)
 
-    final_tmi_df = apply_tmi(df, target_variables, mean_dict)
+    final_tmi_df = apply_tmi(df, lf_target_variables, mean_dict, formtype_long)
 
     final_tmi_df.loc[qa_df.index, "211_trim"] = qa_df["211_trim"]
```

Comment:
(I realise this might not be part of your code) Can we have some comments here to explain this step? It's not obvious to me what's happening.